File: app/src/main/java/com/hanstcs/movieapp/movies/MoviesPresenter.java
package com.hanstcs.movieapp.movies;

import com.hanstcs.movieapp.repository.MoviesRepository;

public class MoviesPresenter implements MoviesContract.Presenter {

    private final MoviesRepository mMoviesRepository;
    private final MoviesContract.View mMoviesView;

    public MoviesPresenter(MoviesContract.View moviesView, MoviesRepository moviesRepository) {
        mMoviesView = moviesView;
        mMoviesRepository = moviesRepository;

        mMoviesView.setPresenter(this);
    }

    @Override
    public void loadMovies() {
        mMoviesView.showMovies(mMoviesRepository.getMovies());
    }
}

File: app/src/main/java/com/hanstcs/movieapp/movies/MoviesPresenter.kt
package com.hanstcs.movieapp.movies

import com.hanstcs.movieapp.data.repository.MoviesRepository

class MoviesPresenter(
        private val moviesView: MoviesContract.View,
        private val moviesRepository: MoviesRepository
) : MoviesContract.Presenter {

    init {
        moviesView.setPresenter(this)
    }

    override fun loadMovies() {
        moviesView.showMovies(moviesRepository.getMovies())
    }
}
